gateway/ln-gateway/src/actor.rs

```diff
@@ -76,53 +81,80 @@ impl GatewayActor {
                     }
                 }
             }
-        });
+        })
+        .await;
 
-        let actor = Self {
+        let mut actor = Self {
             client,
             lnrpc,
-            task_group,
+            task_group: tg,
+            gw_rpc,
+            sender: None,
         };
 
         actor.subscribe_htlcs().await?;
 
         Ok(actor)
     }
 
-    async fn subscribe_htlcs(&self) -> Result<()> {
+    pub async fn stop_subscribing_htlcs(&mut self) -> Result<()> {
+        if let Some(sender) = &self.sender {
+            sender
+                .send(())
```

Comment:
If you're sending `()` you probably want to use something else. like `Arc<Atomicbool>`.

Reply:
Done. What's the reason why? Is there some subtleness using () in channels?